### PR TITLE
New UI for Details Page

### PR DESCRIPTION
**Pull request by Lyczeq**
**Description**
New UI look for the details page. It uses colours related to Kyma's website.

**Reason**
The current UI has been designed to deliver MVP expectations. It doesn't look well.

**Changes:**
- Most feature tiles have their own styling, some of them shares the same one
- Add functions that help in displaying multiple labels, objects etc.
- Add Carousel component from an external library
- Modify Carousel styles

Resolve: #152 
Resolve: #144

### CODE REVIEW

**Comment by Lyczeq** [comment]
We need to discuss in the team which colours should be used and they need to be united.

File: frontend/src/assets/styles/theme.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export const theme = {
   colors: {
     primary: '#0E74DE',
-    secondary: '#de790e',
+    secondary: 'rgb(105,106,209)',
```
Comment (mwisnia97) [inline frontend/src/assets/styles/theme.js:4]: RGB color it's #696AD1 in HEX, I think one convention of color can be better.

File: frontend/src/views/ImageDetailsPage/components/FeatureTiles/LandmarksTile/LandmarksTile.js [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
+const getCoordinates = (cords) => {
+  const latitudeDirection = cords.latitude > 0 ? 'N' : 'S';
+  const longitudeDirection = cords.longitude > 0 ? 'W' : 'E';
+
+  const latitudeValue = `${Math.abs(Number.parseFloat(cords.latitude).toFixed(5))} ${latitudeDirection}`;
+  const longitudeValue = `${Math.abs(Number.parseFloat(cords.longitude).toFixed(5))} ${longitudeDirection} `;
+
+  return `
+        ${latitudeValue}, ${longitudeValue}
+      `;
```
Comment (mwisnia97) [inline frontend/src/views/ImageDetailsPage/components/FeatureTiles/LandmarksTile/LandmarksTile.js:15]: This is an edit from prettier?

File: frontend/src/views/ImageDetailsPage/components/FeatureTiles/TextDocSentimentTile/TextDocSentimentTile.js [hunk 3]
```diff
@@ -1,12 +1,19 @@
 import React from 'react';
-import { FeatureTemplate } from 'assets/styles/style';
+import { FeatureTemplate, FeatureTitle, TextAnalysis } from 'assets/styles/style';
 
 export const TextDocSentimentTile = ({ docSentiment }) => {
+  console.log(docSentiment);
```
Comment (mwisnia97) [inline frontend/src/views/ImageDetailsPage/components/FeatureTiles/TextDocSentimentTile/TextDocSentimentTile.js:5]: Please remove it before merging.

File: frontend/src/views/ImageDetailsPage/components/ImageDetails/ImageDetails.js [hunk 4]
```diff
@@ -46,15 +46,15 @@ const ImageDetails = ({ gcp }) => {
...
+      {textDetails && <FeatureTile title={'Detected words'} features={textDetails.words} offDots/>}
       {logos && <FeatureTile title={'Logos'} features={logos} />}
       {landmarks && <LandmarksTile landmarks={landmarks} />}
       {landmarks && <LandmarksMapTile landmarks={landmarks} />}
       {faceDetails && <FaceTile faceDetails={faceDetails} />}
       {textEntity && <TextEntityTile entity={textEntity} />}
       {textCategory && <TextCategoryTile categories={textCategory} />}
       {textSentiment && <TextDocSentimentTile docSentiment={textSentiment} />}
-      {textSentiment && <TextSentenceSentimentTile sentenceSentiment={textSentiment.sentencetDetails} />}
+      {textSentiment && <TextSentenceSentimentTile sentenceSentiment={textSentiment.sentenceDetails} />}
```
Comment (mwisnia97) [inline frontend/src/views/ImageDetailsPage/components/ImageDetails/ImageDetails.js:57]: Here will be conflict, on main is a newer version.

File: frontend/src/assets/styles/theme.js (hunk 1 above)
Comment (Lyczeq) [reply frontend/src/assets/styles/theme.js:4]: As I have written in the previous comment, the colours need to be unified and we should discuss which of them should be used. That's why most of them aren't in the theme.

File: frontend/src/views/ImageDetailsPage/components/FeatureTiles/LandmarksTile/LandmarksTile.js (hunk 2 above)
Comment (Lyczeq) [reply frontend/src/views/ImageDetailsPage/components/FeatureTiles/LandmarksTile/LandmarksTile.js:15]: No, it's a string

File: frontend/src/assets/styles/style.js [hunk 5]
```diff
@@ -108,19 +109,105 @@ export const Loader = styled.div`
...
+    }
+  }
+
+  .rec-pagination {
+    display: ${({ offDots }) => (offDots ? 'none' : 'flex')};
+  }
+
+  .rec.rec-dot {
+    display: ${({ offDots, isNotMany }) => {
+      if (offDots && isNotMany) return 'none';
```
Comment (Wawrzyn321) [inline frontend/src/assets/styles/style.js:172]: <3

File: frontend/src/assets/styles/style.js [hunk 6]
```diff
@@ -108,19 +109,105 @@ export const Loader = styled.div`
...
+      else if (offDots) return 'none';
+      else if (isNotMany) return 'none';
```
Comment (Wawrzyn321) [inline frontend/src/assets/styles/style.js:174]: Anyway, wouldn't `if (offDots || isNotMany) return 'none';` be enough?

File: frontend/src/views/ImageDetailsPage/components/FeatureTiles/FeatureTile/FeatureTile.js [hunk 7]
```diff
@@ -1,15 +1,46 @@
...
+export const featuresToChunks = (array) => {
+  const coppiedArray = [...array];
+  if (coppiedArray.length < 8) {
+    return {
+      chunks: [coppiedArray],
+      isNotMany: true,
+    };
+  }
+
+  const dividedLength = parseInt(coppiedArray.length / 7, 10);
```
Comment (Wawrzyn321) [inline frontend/src/views/ImageDetailsPage/components/FeatureTiles/FeatureTile/FeatureTile.js:12]: Math.floor?

File: frontend/src/views/ImageDetailsPage/components/FeatureTiles/FeatureTile/FeatureTile.js [hunk 8]
```diff
@@ -1,15 +1,46 @@
-import React from 'react';
-import { FeatureTemplate } from 'assets/styles/style';
+import { FeatureTemplate, FeatureTitle, FeatureCarousel } from 'assets/styles/style';
 
-export const FeatureTile = ({ title, features }) => {
+export const featuresToChunks = (array) => {
+  const coppiedArray = [...array];
```
Comment (Wawrzyn321) [inline frontend/src/views/ImageDetailsPage/components/FeatureTiles/FeatureTile/FeatureTile.js:4]: *copied

File: frontend/src/views/ImageDetailsPage/components/FeatureTiles/FeatureTile/FeatureTile.js [hunk 9]
```diff
@@ -1,26 +1,30 @@
 import { FeatureTemplate, FeatureTitle, FeatureCarousel } from 'assets/styles/style';
 
 export const featuresToChunks = (array) => {
-  const coppiedArray = [...array];
-  if (coppiedArray.length < 8) {
+  const copiedArray = [...array];
+  console.log(copiedArray);
```
Comment (mwisnia97) [inline frontend/src/views/ImageDetailsPage/components/FeatureTiles/FeatureTile/FeatureTile.js:5]: clg

File: frontend/src/views/ImageDetailsPage/components/FeatureTiles/FeatureTile/FeatureTile.js [hunk 10]
```diff
@@ -1,26 +1,30 @@
...
   for (let i = 0; i < dividedLength + 1; i++) {
-    const featuresChunk = coppiedArray.slice(i * helperIterator, displayedFeatures);
+    const featuresChunk = copiedArray.slice(i * helperIterator, displayedFeatures);
     if (featuresChunk.length) {
       result = [...result, featuresChunk];
       displayedFeatures += helperIterator;
     }
   }
+
+  console.log(result);
```
Comment (mwisnia97) [inline frontend/src/views/ImageDetailsPage/components/FeatureTiles/FeatureTile/FeatureTile.js:27]: clg

**Review by mwisnia97 — APPROVED** [review]

**Pull request merged**